src/dymaptic.GeoBlazor.Core/Components/Renderers/RasterStretchRenderer.cs

```diff
@@ -0,0 +1,171 @@
+﻿using dymaptic.GeoBlazor.Core.Components.Layers;
+using dymaptic.GeoBlazor.Core.Components.Renderers.ColorRamps;
+using dymaptic.GeoBlazor.Core.Extensions;
+using dymaptic.GeoBlazor.Core.Serialization;
+using System;
+using System.Buffers.Text;
+using System.Collections.Generic;
+using System.ComponentModel.Design;
+using System.Diagnostics.Contracts;
+using System.Diagnostics.Metrics;
+using System.IO.Pipelines;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading.Tasks;
+
+namespace dymaptic.GeoBlazor.Core.Components.Renderers;
+
+/// <summary>
+///     RasterStretchRenderer defines the symbology with a gradual ramp of colors for each pixel in a ImageryLayer, ImageryTileLayer,
+///     and WCSLayer based on the pixel value. The RasterStretchRenderer works well when you have a large range of values to display,
+///     such as in imagery, aerial photographs, or elevation models. Important to note that RasterStretchRenderer does not inherit from
+///     Renderer class.
+///     <a target="_blank" href="https://developers.arcgis.com/javascript/latest/api-reference/esri-renderers-RasterStretchRenderer.html">
+///         ArcGIS
+///         JS API
+///     </a>
+/// </summary>
+/// 
+public class RasterStretchRenderer : LayerObject
+{
+    /// <summary>
+    ///     Constructor for use in code
+    /// </summary>
+    public RasterStretchRenderer() { }
+
+    public RasterStretchRenderer(MultipartColorRamp? colorRamp = null, bool? computeGamma = null, bool? dynamicRangeAdjustment = null, List<int>? gamma = null, int? outputMax = null, int? outputMin = null,
+        StretchType? stretchType = null, List<List<int>>? statistics = null, bool? useGamma = null, int? numberOfStandardDeviations = null)
+    {
+        ColorRamp = colorRamp;
+        ComputeGamma = computeGamma;
+        DynamicRangeAdjustment = dynamicRangeAdjustment;
+        Gamma = gamma;
+        OutputMax = outputMax;
+        OutputMin = outputMin;
+        StretchType = stretchType;
+        Statistics = statistics;
+        UseGamma = useGamma;
+        NumberOfStandardDeviations = numberOfStandardDeviations;
+    }
+
+    /// <inheritdoc />
+    [JsonPropertyName("type")]
+    public string Type => "raster-stretch";
+
+    /// <summary>
+    ///     The stretched values are mapped to this specified color ramp.
+    /// </summary>
+    public MultipartColorRamp? ColorRamp { get; private set; }
+
+    /// <summary>
+    ///     The computeGamma automatically calculates best gamma value to render exported image based on empirical model. This is applicable to any stretch type when useGamma is true.
+    /// </summary>
+    public bool? ComputeGamma { get; private set; }
```

Comment:
Everything not set via `RegisterChildComponent` should be a `[Parameter]`

Reply:
corrected, once the constructor was refactored, these should have been corrected.